// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/compiler/overloadsWithComputedNames.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -14,19 +14,23 @@
 declare const uniqueSym2: unique symbol;
 declare const sym: symbol;
 declare const strUnion: 'foo' | 'bar';
 declare class C1 {
+    [sym](): void;
     [uniqueSym2](): void;
     [uniqueSym](): void;
 }
 interface I1 {
+    [sym](): void;
     [uniqueSym2](): void;
     [uniqueSym](): void;
     [uniqueSym](): void;
 }
 declare class C2 {
+    [strUnion](): void;
 }
 declare class I2 {
+    [strUnion](): void;
 }
 declare class C3 {
     [1](): void;
     [2](): void;
@@ -43,21 +47,16 @@
 overloadsWithComputedNames.ts(8,9): error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations.
 overloadsWithComputedNames.ts(14,5): error TS2391: Function implementation is missing or not immediately following the declaration.
 overloadsWithComputedNames.ts(16,5): error TS2389: Function implementation name must be '["bar"]'.
 overloadsWithComputedNames.ts(28,5): error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-overloadsWithComputedNames.ts(28,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 overloadsWithComputedNames.ts(29,5): error TS2391: Function implementation is missing or not immediately following the declaration.
 overloadsWithComputedNames.ts(35,5): error TS1169: A computed property name in an interface must refer to an expression whose type is a literal type or a 'unique symbol' type.
 overloadsWithComputedNames.ts(42,5): error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-overloadsWithComputedNames.ts(42,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
-overloadsWithComputedNames.ts(43,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 overloadsWithComputedNames.ts(47,5): error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-overloadsWithComputedNames.ts(47,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
-overloadsWithComputedNames.ts(48,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 overloadsWithComputedNames.ts(52,5): error TS2391: Function implementation is missing or not immediately following the declaration.
 
 
-==== overloadsWithComputedNames.ts (15 errors) ====
+==== overloadsWithComputedNames.ts (10 errors) ====
     // https://github.com/microsoft/TypeScript/issues/52329
     class Person {
         ["B"](a: number): string;
         ["A"](a: string|number): number | string {
@@ -95,10 +94,8 @@
     class C1 {
         [sym](): void;  // should error
         ~~~~~
 !!! error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-        ~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         [uniqueSym2](): void;   // should error
         ~~~~~~~~~~~~
 !!! error TS2391: Function implementation is missing or not immediately following the declaration.
         [uniqueSym](): void;
@@ -117,24 +114,16 @@
     class C2 {
         [strUnion](): void; // should error
         ~~~~~~~~~~
 !!! error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-        ~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         [strUnion]() { }
-        ~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     class I2 {
         [strUnion](): void; // should error
         ~~~~~~~~~~
 !!! error TS1168: A computed property name in a method overload must refer to an expression whose type is a literal type or a 'unique symbol' type.
-        ~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
         [strUnion]() { }
-        ~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
     }
     
     class C3 {
         [1](): void;  // should error
